core/src/main/java/org/apache/iceberg/avro/AvroFileAppender.java

```diff
@@ -78,14 +78,13 @@ public Metrics metrics() {
 
   @Override
   public long length() {
-    if (stream != null) {
-      try {
-        return stream.getPos();
-      } catch (IOException e) {
-        throw new RuntimeIOException(e, "Failed to get stream length");
-      }
+    Preconditions.checkNotNull(stream, "Failed to get stream length: no open stream");
```

Comment:
Yes, I changed this to `IllegalStateException`  based on Dan's comment last time.
https://github.com/apache/iceberg/pull/6887#discussion_r1114846404

Reply:
did you potentially forget to push? Because I don't see `Preconditions.checkState(...)` here

Reply:
Aah, I used `checkNotNull` before instead of `checkState` before as it was just replacing NULL check. 

But yeah. It is better to throw  `IllegalStateException` instead of `NPE`.  I have updated now.